README.md

```diff
@@ -24,18 +24,106 @@ Then `bundle exec gruf` and you'll automatically have prometheus metrics for you
+|ruby_grpc_pool_initial_size|gauge|Initial size of the gRPC thread pool|
+|ruby_grpc_poll_period|gauge|Polling period for the gRPC thread pool|
+
+### Server Metrics
+
+Furthermore, the server interceptor measures general counts (and optionally, latencies), and can be setup via:
+
+```ruby
+::Gruf.interceptors.use(::Gruf::Prometheus::Server::Interceptor)
+```
+
+This will output the following metrics:
+
+|Name|Type|Description|
+|---|---|---|
+|ruby_grpc_server_started_total|counter|Total number of RPCs started on the server|
+|ruby_grpc_server_handled_total|counter|Total number of RPCs completed on the server, regardless of success or failure|
+|ruby_grpc_server_handled_latency_seconds|histogram|Histogram of response latency of RPCs handled by the server, in seconds|
+
+Note that the histogram is disabled by default - you'll have to turn it on either through the `server_measure_latency`
+configuration yielded in `Gruf::Prometheus.configure`, or through the `PROMETHEUS_SERVER_MEASURE_LATENCY` environment
+variable. Also, the `measure_latency: true` option can be passed as a second argument to `Gruf.interceptors.use` to 
+configure this directly in the interceptor.
+
+### Client Metrics
+
+gruf-prometheus can also measure gruf client timings, via the interceptor:
+
+```ruby
+Gruf::Client.new(
+  service: MyService,
+  client_options: {
+    interceptors: [Gruf::Prometheus::Client::Interceptor.new]
+  }
+)
+``` 
+
+|Name|Type|Description|
+|---|---|---|
+|ruby_grpc_client_started_total|counter|Total number of RPCs started by the client|
+|ruby_grpc_client_completed|counter|Total number of RPCs completed by the client, regardless of success or failure|
+|ruby_grpc_client_completed_latency_seconds|histogram|Histogram of response latency of RPCs completed by the client, in seconds|
+
+Note that the histogram is disabled by default - you'll have to turn it on either through the `client_measure_latency`
+configuration yielded in `Gruf::Prometheus.configure`, or through the `PROMETHEUS_CLIENT_MEASURE_LATENCY` environment
+variable. Optionally, you can pass in `measure_latency: true` into the Interceptor directly as an option argument in the 
+initializer.
+
+### Running the Client Interceptor in Non-gRPC Processes
+
+One caveat is that you _must_ have the appropriate Type Collector setup in whatever process you are running in. If
+you are already doing this in a gruf gRPC service that is using the hook provided by this gem above, no further 
+configuration is needed. Otherwise, in whatever bc-prometheus-ruby configuration you have setup, you'll need to ensure
+the type collector is loaded: 
+
+```ruby
+# prometheus_server is whatever `::Bigcommerce::Prometheus::Server` instance you are using in the current process
+# Often hooks into these are exposed as configuration options, e.g. `web_collectors`, `resque_collectors`, etc
+prometheus_server.add_type_collector(::Gruf::Prometheus::Client::TypeCollector.new)  
```

Comment:
If you're using gruf-prometheus on a gRPC process, with `Gruf.hooks.use(Gruf::Prometheus::Hook)`, then you don't need to add the Type Collector explicitly (the hook does this). Otherwise, you need to add it manually to whatever instrumentation setup you have running.